# Only use FFmpeg AV1 codec if the decoder is enabled
#
# This fixes building with system FFmpeg libraries without AV1 codec support because
# in version 64, AV1 decoder is disabled by default.

--- a/media/ffmpeg/ffmpeg_common.cc
+++ b/media/ffmpeg/ffmpeg_common.cc
@@ -189,8 +189,10 @@
       return kCodecVP8;
     case AV_CODEC_ID_VP9:
       return kCodecVP9;
+#if BUILDFLAG(ENABLE_AV1_DECODER)
     case AV_CODEC_ID_AV1:
       return kCodecAV1;
+#endif
     default:
       DVLOG(1) << "Unknown video CodecID: " << codec_id;
   }
@@ -213,8 +215,10 @@
       return AV_CODEC_ID_VP8;
     case kCodecVP9:
       return AV_CODEC_ID_VP9;
+#if BUILDFLAG(ENABLE_AV1_DECODER)
     case kCodecAV1:
       return AV_CODEC_ID_AV1;
+#endif
     default:
       DVLOG(1) << "Unknown VideoCodec: " << video_codec;
   }
@@ -498,10 +502,12 @@
       format = PIXEL_FORMAT_YV12;
       profile = VP9PROFILE_PROFILE0;
       break;
+#if BUILDFLAG(ENABLE_AV1_DECODER)
     case kCodecAV1:
       format = PIXEL_FORMAT_I420;
       profile = AV1PROFILE_PROFILE0;
       break;
+#endif
 #if BUILDFLAG(ENABLE_HEVC_DEMUXING)
     case kCodecHEVC:
       profile = HEVCPROFILE_MAIN;
